core/templates/dev/head/exploration_editor/preview_tab/PreviewTab.js

```diff
@@ -42,17 +123,21 @@ oppia.controller('PreviewTab', [
       $timeout(function() {
         oppiaPlayerService.populateExploration({
           category: explorationCategoryService.savedMemento,
-          init_state_name: explorationInitStateNameService.savedMemento,
+          init_state_name: stateName,
           param_changes: explorationParamChangesService.savedMemento,
           param_specs: explorationParamSpecsService.savedMemento,
           states: explorationStatesService.getStates(),
           title: explorationTitleService.savedMemento,
           skin_customizations: {
             panels_contents: explorationGadgetsService.getPanelsContents()
           }
-        }, []);
+        }, manualParamChanges);
         $scope.isExplorationPopulated = true;
       }, 200);
+    };
+
+    $scope.$on('updateActiveState', function(evt, stateName) {
```

Comment:
This is intended to make sure the active state is kept up-to-date when navigating the exploration in preview mode, either by clicking through to the next card, or by clicking on the progress dots.  That way, when clicking back to the editor, the state doesn't change.  This wasn't listed as one of the milestones in #921, but it was mentioned as desirable behaviour in the discussion.  Personally, I think it is intuitive to always stay in the same state when switching between the editor and preview.